ui/app/AppLayouts/Wallet/HistoryTab.qml

```diff
@@ -129,100 +132,107 @@ Item {
-                    text: to
-                    width: 100
-                    elide: Text.ElideMiddle
-                    anchors.right: parent.right
-                    anchors.top: parent.top
-                    font.pixelSize: 15
+                    text: isIncoming ? fromAddress : to
+                    maxWidth: 120
+                    width: 120
+                    horizontalAlignment: Text.AlignRight
+                    font.pixelSize: Style.current.primaryTextFontSize
+                    color: Style.current.textColor
                 }
             }
 
-            Item {
+            Row {
                 id: timeInfo
                 anchors.right: parent.right
+                anchors.rightMargin: Style.current.smallPadding
                 anchors.top: parent.top
                 anchors.topMargin: Style.current.bigPadding
-                width: children[0].width + children[1].width + children[2].width
+                spacing: 5
 
                 StyledText {
-                    text: "• "
+                    text: " • "
                     font.weight: Font.Bold
-                    anchors.right: timeIndicator.left
                     color: Style.current.secondaryText
-                    anchors.top: parent.top
-                    font.pixelSize: 15
+                    font.pixelSize: Style.current.primaryTextFontSize
                 }
 
                 StyledText {
                     id: timeIndicator
-                    text: "At "
-                    anchors.right: timeValue.left
+                    text: qsTr("At ")
                     color: Style.current.secondaryText
-                    anchors.top: parent.top
-                    font.pixelSize: 15
+                    font.pixelSize: Style.current.primaryTextFontSize
                     font.strikeout: false
                 }
-
                 StyledText {
                     id: timeValue
-                    text: timestamp
-                    anchors.right: parent.right
-                    anchors.top: parent.top
-                    font.pixelSize: 15
+                    text: new Date(timestamp).toLocaleString(globalSettings.locale)
+                    font.pixelSize: Style.current.primaryTextFontSize
                     anchors.rightMargin: Style.current.smallPadding
                 }
             }
         }
     }
 
+    StyledText {
+        id: nonArchivalNodeError
+        visible: walletModel.isNonArchivalNode
+        height: visible ? 20 : 0
+        anchors.top: parent.top
+        text: qsTr("Status Desktop is connected to a non-archival node. Transaction history may be incomplete.")
+        font.pixelSize: Style.current.primaryTextFontSize
+        color: Style.current.danger
+    }
+
+    StyledText {
+        id: noTxs
+        anchors.top: nonArchivalNodeError.bottom
+        visible: transactionListRoot.count === 0
+        height: visible ? 36 : 0
```

Comment:
I see now that you put different heights to create spacing. If you want, you can leave it as is, but it's probably better to handle the spacing with the anchors margins.
```suggestion
        height: visible ? implicitHeight : 0
```